core/includes/actions.inc

```diff
@@ -73,16 +73,16 @@ function actions_get_info($action_name = NULL) {
  *
  * @param string $action_name
  *   The machine name of the action to be executed.
- * @param Entity $entity
+ * @param ?Entity $entity
  *   The entity object that the action will act upon, such as a node, user, or
  *   comment object.
  *
- * @return
+ * @return mixed
  *   The result of the executed action callback.
  *
  * @ingroup actions
  */
-function actions_execute($action_name, Entity $entity = NULL, &$context) {
+function actions_execute($action_name, ?Entity $entity, &$context) {
```

Comment:
I'm guessing this still needs `= NULL` for BC.

Reply:
PHP 8.4 requires that optional parameters follow required ones. In this case `&$context` is a required parameter, so all calling functions need to pass in `NULL` or an `$entity` value already, because they also need to pass in `$context`, otherwise you'd get a missing argument PHP error. So this is a safe change to make.